### PR TITLE
Add blog for microshift preset

### PR DESCRIPTION
**Pull request by praveenkumar**

### CODE REVIEW

**Comment by praveenkumar** [comment]
/hold

Till 2.16 release happen.

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+title: " Introducing MicroShift preset for CRC"
+date: 2023-04-05T07:12:10+05:30
+---
+
+Are you a developer who wants to use OpenShift Kubernetes but doesn't have a powerful system?
+Or do you need to deploy to a small form factor or edge computing environment? 
+If so, you'll be happy to hear about [MicroShift](https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/getting_started/index),
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md]: about Microshift -> about our new MicroShift preset ?

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Currently, OpenShift preset provided by CRC is required 9GB of RAM and 4CPUs.
+With the release of CRC-2.16, users can now use the MicroShift preset, which only requires 4GB of RAM and 2 CPUs.
+
+What makes MicroShift different from OpenShift Kubernetes? MicroShift is designed for small form factor
+and edge computing environments, which means it's optimized for lower resource requirements. MicroShift
+uses a subset of the OpenShift Kubernetes components and removes some of the features that aren't needed in these environments.
+
+[Key differences from OpenShift Kubernetes](https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/getting_started/microshift-architecture#microshift-differences-oke_microshift-architecture)
+
+To get started with MicroShift, you'll need to download and install the [latest version]() of CRC.
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md]: link is missing.

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+If you already have a CRC instance, delete it using the `crc delete` command. 
+Then set the preset to MicroShift using the `crc config set preset microshift` command.
+Finally, set up and start CRC using the `crc setup` and `crc start` commands.
+
+After you start CRC, you'll receive the `kubeconfig` file location, which you can export to connect with the cluster.
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md]: After CRC is started, a `kubeconfig` file will be generated, which is used to connect to the cluster.

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+If you already have a CRC instance, delete it using the `crc delete` command. 
+Then set the preset to MicroShift using the `crc config set preset microshift` command.
+Finally, set up and start CRC using the `crc setup` and `crc start` commands.
+
+After you start CRC, you'll receive the `kubeconfig` file location, which you can export to connect with the cluster. 
+
+To try out MicroShift, we've created a sample demo repository that you can find on [GitHub](https://github.com/praveenkumar/simple-go-server).
+It includes a workload that you can use to test out the MicroShift preset.
+
+We'd love to hear your feedback and suggestions on the MicroShift preset. Please let us know in our [release discussion](), or if you encounter any issues, please file them on our [issue tracker](https://github.com/crc-org/crc/issues).
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md]: missing link.

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+In conclusion, MicroShift preset is a great option for users  who need to use OpenShift Kubernetes in resource-limited environments without some of the specific features.
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md]: extra space between "users" and "who".
"some of the specific features"? Which features? Without some of the features of a Red Hat OpenShift cluster?

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+date: 2023-04-05T07:12:10+05:30
+---
+
+Are you a developer who wants to use OpenShift Kubernetes but doesn't have a powerful system?
+Or do you need to deploy to a small form factor or edge computing environment? 
+If so, you'll be happy to hear about [MicroShift](https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/getting_started/index),
+a lightweight version of OpenShift Kubernetes that's optimized for resource-limited environments.
+
+Currently, OpenShift preset provided by CRC is required 9GB of RAM and 4CPUs.
+With the release of CRC-2.16, users can now use the MicroShift preset, which only requires 4GB of RAM and 2 CPUs.
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md]: CRC-2.16 - > CRC 2.16
users -> you ?

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+title: " Introducing MicroShift preset for CRC"
+date: 2023-04-05T07:12:10+05:30
+---
+
+Are you a developer who wants to use OpenShift Kubernetes but doesn't have a powerful system?
+Or do you need to deploy to a small form factor or edge computing environment? 
+If so, you'll be happy to hear about [MicroShift](https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/getting_started/index),
+a lightweight version of OpenShift Kubernetes that's optimized for resource-limited environments.
+
+Currently, OpenShift preset provided by CRC is required 9GB of RAM and 4CPUs.
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md]: The OpenShift preset currently provided by CRC requires 9GB of RAM and 4 CPUs

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+To get started with MicroShift, you'll need to download and install the [latest version]() of CRC.
+If you already have a CRC instance, delete it using the `crc delete` command. 
+Then set the preset to MicroShift using the `crc config set preset microshift` command.
+Finally, set up and start CRC using the `crc setup` and `crc start` commands.
+
+After you start CRC, you'll receive the `kubeconfig` file location, which you can export to connect with the cluster. 
+
+To try out MicroShift, we've created a sample demo repository that you can find on [GitHub](https://github.com/praveenkumar/simple-go-server).
+It includes a workload that you can use to test out the MicroShift preset.
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md]: I was thinking we could also provide an image that can be directly deployed on the cluster to make sure it works, and then to provide instructions how to build that image for those that are interested in this.

Comment (praveenkumar) [reply content/posts/2023-04-05-microshift-on-crc.md]: that something we can do that on demo repo if we need to update it. One think I have in mind to add step how microshift preset can be added as github action.

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+Are you a developer who wants to use OpenShift Kubernetes but doesn't have a powerful system?
+Or do you need to deploy to a small form factor or edge computing environment? 
+If so, you'll be happy to hear about our new [MicroShift](https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/getting_started/index) preset,
+a lightweight version of OpenShift Kubernetes that's optimized for resource-limited environments.
+
+The OpenShift preset currently provided by CRC requires 9GB of RAM and 4 CPUs.
+With the release of CRC 2.16, you can now use the MicroShift preset, which only requires 4GB of RAM and 2 CPUs.
+
+What makes MicroShift different from OpenShift Kubernetes? MicroShift is designed for small form factor
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md:14]: They call it "OpenShift Kubernetes Engine" in the document which is linked  below.

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
+---
+title: " Introducing MicroShift preset for CRC"
+date: 2023-04-05T07:12:10+05:30
+---
+
+Are you a developer who wants to use OpenShift Kubernetes but doesn't have a powerful system?
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md]: They call it "OpenShift Kubernetes Engine" in the official document which is linked  below. I would replace it everywhere in that post.
I don't know if this should be "Red Hat OpenShift Kubernetes Engine" the first time it's mentioned.

File: content/posts/2023-04-05-microshift-on-crc.md (hunk 8 above)
Comment (cfergeau) [reply content/posts/2023-04-05-microshift-on-crc.md]: "It includes a workload ..." is misleading if we don't provide it :) "It includes step by step instructions to build and deploy a workload on a CRC instance using the MicroShift preset"

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+The OpenShift preset currently provided by CRC requires 9GB of RAM and 4 CPUs.
+With the release of CRC 2.16, you can now use the MicroShift preset, which only requires 4GB of RAM and 2 CPUs.
+
+What makes MicroShift different from OpenShift Kubernetes? MicroShift is designed for small form factor
+and edge computing environments, which means it's optimized for lower resource requirements. MicroShift
+uses a subset of the OpenShift Kubernetes components and removes some of the features that aren't needed in these environments.
+
+[Key differences from OpenShift Kubernetes](https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/getting_started/microshift-architecture#microshift-differences-oke_microshift-architecture)
+
+To get started with MicroShift, you'll need to download and install the [latest version](https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/crc/latest) of CRC.
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md:20]: It's missing a mention of the workaround.

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+and edge computing environments, which means it's optimized for lower resource requirements. MicroShift
+uses a subset of the OpenShift Kubernetes Engine components and removes some of the features that aren't needed in these environments.
+
+[Key differences from OpenShift Kubernetes Engine](https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/getting_started/microshift-architecture#microshift-differences-oke_microshift-architecture)
+
+To get started with MicroShift, you'll need to download and install the [latest version](https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/crc/latest) of CRC.
+If you already have a CRC instance, delete it using the `crc delete` command.
+Then set the preset to MicroShift using the `crc config set preset microshift` command.
+Finally, set up and start CRC using the `crc setup` and `crc start` commands.
+In case you want to use `crc oc-env` command, do check the [workaround](https://github.com/crc-org/crc/issues/3581#issuecomment-1497174956)
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md]: Maybe "In order to use `oc` and `crc oc-env` with crc version 2.16, a [workaround] is needed" ?

Comment (praveenkumar) [reply content/posts/2023-04-05-microshift-on-crc.md]: > In order to use oc

Not `oc`, which works without any issue if user have it in the path. it is just the `oc-env` which have problem for creating the symlink to path.

Comment (cfergeau) [reply content/posts/2023-04-05-microshift-on-crc.md]: Maybe "In order to use crc oc-env with crc version 2.16, a [workaround] is needed" ?

File: content/posts/2023-04-05-microshift-on-crc.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+and edge computing environments, which means it's optimized for lower resource requirements. MicroShift
+uses a subset of the OpenShift Kubernetes Engine components and removes some of the features that aren't needed in these environments.
+
+[Key differences from OpenShift Kubernetes Engine](https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/getting_started/microshift-architecture#microshift-differences-oke_microshift-architecture)
+
+To get started with MicroShift, you'll need to download and install the [latest version](https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/crc/latest) of CRC.
+If you already have a CRC instance, delete it using the `crc delete` command.
+Then set the preset to MicroShift using the `crc config set preset microshift` command.
+Finally, set up and start CRC using the `crc setup` and `crc start` commands.
+With crc version 2.16, In order to use the oc binary from `crc oc-env`, a [workaround](https://github.com/crc-org/crc/issues/3581#issuecomment-1497174956) is needed.
```
Comment (cfergeau) [inline content/posts/2023-04-05-microshift-on-crc.md]: `, in order` without the capital letter, apart from this, ACK.

**Pull request merged**